cuda_core/tests/test_program.py

```diff
@@ -8,10 +8,19 @@
 
 import pytest
 
-from cuda.core.experimental import Program
+from cuda import cuda, nvrtc
+from cuda.core.experimental import Device, Program
 from cuda.core.experimental._module import Kernel, ObjectCode
 
 
+def can_load_generated_ptx():
+    _, driver_ver = cuda.cuDriverGetVersion()
+    _, nvrtc_major, nvrtc_minor = nvrtc.nvrtcVersion()
+    if nvrtc_major * 1000 + nvrtc_minor * 10 > driver_ver:
+        return False
+    return True
```

Comment:
FYI @ksimpson-work this is akin to this snippet that I added to CuPy in the past (PTX might not be loadable/JIT'able if it's newer than the driver):
https://github.com/cupy/cupy/blob/8eb16ac910e85c119a20f68a69de9a2e6034069c/tests/cupy_tests/core_tests/test_raw.py#L557-L568